config.cli.js

```diff
@@ -36,7 +36,7 @@ const {argv} = require('yargs')
   })
   .option('auth', {
     describe: 'Access control and user management method',
-    choices: ['default', 'none'],
+    choices: ['default', 'httpbasic', 'none'],
```

Comment:
`header` would be ambigous: what if tomorrow we add http digest, or any other bearer torken?

Reply:
With logics of this PR, any header authentication method shall only take the username. So if digest method needs to be supported, we simply parse the username from that type of header without changing other logics. Headers which don't supply username such as `Bearer` can't be supported.

Reply:
the logic is executed only when type `httpbasic`, and in case of digest the authorization header will be parsed differently also.

`Bearer` could provide the username in the payload, and we will have a method `bearerSomething` and add a branch in the switch.

or the bearer could provided metadata needed to extract the username for in other way and finally compare with the jwt token username, adding extra logic.

that's why reducing everything to `header` method it is not the correct solution in my opinion.

Reply:
For all other types which username can be parsed, the authentication and user management logics are the same. The only difference is how to parse the username which can be handled by a simple switch case. I prefer that we use `header` to be more general.

Reply:
> The only difference is how to parse the username which can be handled by a simple switch case. I prefer that we use header to be more general.

that's what currently happens: a switch case on the `auth` value for different logic how to parse the username.
if we have a generic `header` value the switch should need extra information (we could have a `headertype` settings, it won't change much indeed). if we don't have this extra information we have to inspect the header and apply the parse method that will apply. it would be nice, but I doubt that there could be the cases where the same header inspection won't give enough information on how to actually parse it: then I don't see any other solution than applying multiple ambigous parsing until one won't succed.

what do you think?

Reply:
For now, we can enforce the `Basic` type.

Reply:
so auth value `header` that will assume only `Basic` type?
then it's only a matter of replacing `httpbasic` with `header`

I'm still not convinved this is the best way to easy future support to different methods, but as you prefer

Reply:
Yeah. It would be very straightforward to add support for other auth header types in the future. After all, we only want the username.